python_modules/dagster/dagster/core/definitions/solid_definition.py

```diff
@@ -137,61 +138,80 @@ def __init__(
         )
 
     def __call__(self, *args, **kwargs) -> Any:
-        from .composition import is_in_composition
-        from .decorators.solid import DecoratedSolidFunction
-        from ..execution.context.invocation import UnboundSolidExecutionContext
+        from .composition import (
+            is_in_composition,
+            enter_op_execution_context,
+            is_in_op_execution_context,
+            current_op_execution_context,
+            exit_op_execution_context,
+        )
 
         if is_in_composition():
             return super(SolidDefinition, self).__call__(*args, **kwargs)
+        elif is_in_op_execution_context():
+            name, source = current_op_execution_context()
+            raise DagsterInvalidInvocationError(
+                f"Attempted to invoke @{self.node_type_str} '{self.name}' within {source} '{name}', which is undefined behavior. In order to compose invocations, check out the graph API: https://docs.dagster.io/concepts/ops-jobs-graphs/nesting-graphs#nesting-graphs"
+            )
```

Comment:
This should probably mention `invoke` method explicitly or how else would one discover it? Maybe update the linked docs as well with a section 

A little worried about this deep link breaking over time - not sure best way to defend against that.